src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -320,19 +340,184 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
+
+                try
+                {
+                    // Store file as blob.
+                    _dataRepository.CreateDataInStorage(theStream, newDataElement.StorageUrl);
+
+                    // Add file to instance.
+                    storedInstance.Data.Add(newDataElement);
+
+                    // Update instance.
+                    storedInstance = _instanceRepository.Update(storedInstance).Result;
+                }
+                catch (Exception e)
+                {
+                    errorResult = StatusCode(500, $"Unable to create instance data in storage: {e}");
+                }
+
+                section = reader.ReadNextSectionAsync().Result;
+            }
+
+            return storedInstance;
+        }
+
+        /// <summary>
+        /// Method to read the instance object from the HttpRequest body. 
+        /// </summary>
+        /// <param name="request">The HttpRequest</param>
+        /// <param name="appInfo">The application information</param>
+        /// <returns>The instance object</returns>
+        private async Task<Instance> ReadInstanceTemplateFromBody(HttpRequest request, Application appInfo)
+        {
+            Instance instanceTemplate = null;
+
+            if (MultipartRequestHelper.IsMultipartContentType(request.ContentType))
+            {
+                // Only read the first section of the mulitpart message.
+                MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(request.ContentType);
+                string boundary = MultipartRequestHelper.GetBoundary(mediaType, _defaultFormOptions.MultipartBoundaryLengthLimit);
+
+                MultipartReader reader = new MultipartReader(boundary, request.Body);
+                MultipartSection section = reader.ReadNextSectionAsync().Result;
+
+                bool hasContentDispositionHeader =
+                        ContentDispositionHeaderValue.
+                        TryParse(section.ContentDisposition, out ContentDispositionHeaderValue contentDisposition);
+
+                if (hasContentDispositionHeader)
+                {
+                    // Check if the content disposition name is "instance".
+                    if (contentDisposition.Name.Value == "instance")
+                    {
+                        string contentType = section.ContentType.Split(";")[0];
+
+                        // Check if the content type of the multipart section is declared for the application (for example "application/json").
+                        if (!appInfo.ElementTypes.Exists(e => e.AllowedContentType.Contains(contentType)))
+                        {
+                            instanceTemplate = JsonConvert.DeserializeObject<Instance>(await section.ReadAsStringAsync());
+                            request.Body.Position = 0;
```

Comment:
This is only done if first form-data in the multipart is called instance. If instance is not present you should reset the stream position. Move this to the end of the multipart check.

Reply:
Fixed.